src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -307,7 +307,7 @@ public function setMethodCalls(array $calls = []): static
     {
         $this->calls = [];
         foreach ($calls as $call) {
-            $this->addMethodCall($call[0], $call[1], $call[2] ?? false);
+            $this->addMethodCall($call[0], $call[1], isset($call[2]) && $call[2]);
```

Comment:
Do you have stack trace for the case triggering the deprecation here (without your patch) ? Maybe we have some tests not using the expected structure for the array, but it might be better to fix the root cause instead of a symptom (replacing the boolean type juggling in function context by a boolean type juggling in boolean operator context, which is not deprecated)